worlds/ff1/Client.py

```diff
@@ -0,0 +1,298 @@
+              "Gold160", "Gold180", "Gold240", "Gold255", "Gold260", "Gold295", "Gold300", "Gold315", "Gold330",
+              "Gold350", "Gold385", "Gold400", "Gold450", "Gold500", "Gold530", "Gold575", "Gold620", "Gold680",
+              "Gold750", "Gold795", "Gold880", "Gold1020", "Gold1250", "Gold1455", "Gold1520", "Gold1760", "Gold1975",
+              "Gold2000", "Gold2750", "Gold3400", "Gold4150", "Gold5000", "Gold5450", "Gold6400", "Gold6720",
+              "Gold7340", "Gold7690", "Gold7900", "Gold8135", "Gold9000", "Gold9300", "Gold9500", "Gold9900",
+              "Gold10000", "Gold12350", "Gold13000", "Gold13450", "Gold14050", "Gold14720", "Gold15000", "Gold17490",
+              "Gold18010", "Gold19990", "Gold20000", "Gold20010", "Gold26000", "Gold45000", "Gold65000"]
+
+extended_consumables = ["Smoke", "FullCure", "Blast", "Phoenix",
+                        "Flare", "Black", "Refresh", "Guard",
+                        "Wizard", "HighPotion", "Cloak", "Quick"]
+
+ext_consumables_lookup = {"Smoke": "Ext1", "FullCure": "Ext2", "Blast": "Ext3", "Phoenix": "Ext4",
+                          "Flare": "Ext1", "Black": "Ext2", "Refresh": "Ext3", "Guard": "Ext4",
+                          "Wizard": "Ext1", "HighPotion": "Ext2", "Cloak": "Ext3", "Quick": "Ext4"}
+
+ext_consumables_locations = {"Ext1": 0x3C, "Ext2": 0x3D, "Ext3": 0x3E, "Ext4": 0x3F}
+
+
+movement_items = ["Ship", "Bridge", "Canal", "Canoe"]
+
+no_overworld_items = ["Sigil", "Mark"]
+
+
+class FF1Client(BizHawkClient):
+    game = "Final Fantasy"
+    system = "NES"
+
+    def __init__(self):
+        self.wram = "RAM"
+        self.sram = "WRAM"
+        self.rom = "PRG ROM"
+        self.consumable_stack_amounts = None
+        self.weapons_queue = deque()
+        self.armor_queue = deque()
+
+
+    async def validate_rom(self, ctx: "BizHawkClientContext") -> bool:
+        try:
+            # Check ROM name/patch version
+            rom_name = ((await bizhawk.read(ctx.bizhawk_ctx, [(rom_name_location, 0x0D, self.rom)]))[0])
+            rom_name = rom_name.decode("ascii")
+            if rom_name != "FINAL FANTASY":
+                return False  # Not a Final Fantasy 1 ROM
+        except bizhawk.RequestFailedError:
+            return False  # Not able to get a response, say no for now
+
+        ctx.game = self.game
+        ctx.items_handling = 0b111
+        ctx.want_slot_data = True
+
+        return True
+
+    async def game_watcher(self, ctx: "BizHawkClientContext") -> None:
+        if ctx.server is None:
+            return
+
+        if ctx.slot is None:
+            return
+        try:
+            if await self.check_status_okay_to_process(ctx):
+                if self.consumable_stack_amounts is None:
+                    self.consumable_stack_amounts = {}
+                    self.consumable_stack_amounts["Shard"] = 1
+                    self.consumable_stack_amounts["Tent"] = (await self.read_rom(ctx, 0x47400, 1))[0] + 1
```

Comment:
I don't know if BizhawkClient supports switching between seeds and worlds like SNIClient does.
But if someone switches from one FF1 world to another, this looks like it might still use the `consumable_stack_amounts` from the first world while playing the second.